client/src/client/sessions_state_machine.py

```diff
@@ -189,17 +196,17 @@ def mark_outbound_session_active(self, message: CansMessage) -> None:
         # Run decryption to have the session well established at
         # olm level - without this, olm will continue producing
         # prekey messages
-        olm_message = OlmMessage(message.payload)
+        olm_message = OlmMessage(message.payload["magic"])
```

Comment:
Is this "magic" only used to authenticate session at Olm level? Can't we just use "text" field? Does it have any other purpose?

Reply:
^Payload could then be a namedTuple(text, "cookie", target), which I think would be easier to use

Reply:
Depending on message ID payload can have different structure. It cannot be typed based on user messages alone, we have other traffic also, where there are more fields in the payload. I could revert back to using message.payload of type string for the handshake messages (remove the indirection via the "magic" field)